src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/index.jsx

```diff
@@ -14,6 +15,20 @@ export default class OrganizationDeveloperSettings extends AsyncView {
     return [['applications', `/organizations/${orgId}/sentry-apps/`]];
   }
 
+  removeApp = app => {
+    const api = this.api;
+    const origApps = this.state.applications;
+    const apps = this.state.applications.filter(a => a.slug !== app.slug);
+    removeSentryApp(api, app).then(
+      () => {
+        this.setState({applications: apps});
+      },
+      () => {
+        this.setState({applications: origApps});
```

Comment:
Do we actually have to do anything here? `state.applications` will be unchanged unless we reach the success function.